hw3/01-routing.js HW3

```diff
@@ -0,0 +1,76 @@
+const http = require("http");
+
+const server = http.createServer((req, res) => {
+  //if the url is /welcome
+  if (req.url === "/welcome") {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write("<h1>Welcome to this page</h1>");
+    res.end();
+  }
+  // if the url is /redirect
+  else if (req.url === "/redirect") {
+    res.writeHead(302, { Location: "/redirected" });
+    console.log("redirected");
+    return res.end();
+  } else if (req.url === "/redirected") {
+    console.log("page redirected");
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write("<h1>Redirected from /redirect</h1>");
+    res.end();
+  }
+  //if the url is/cache
+  else if (req.url === "/cache") {
+    res.writeHead(200, {
+      "Content-Type": "text/html",
+      "Cache-control": "max-age=86400",
+    });
+    res.write("<h1>This resource was cached</h1>");
+    res.end();
+  }
+
+  //if the url is /cookie
+  else if (req.url === "/cookie") {
+    res.write("Cookies... Yummm");
+    res.writeHead(200, {
+      "Content-Type": "text/plain",
+      "Set-Cookie": "hello=world",
+    });
+
+    res.end();
+  }
+  //if the url is /check-cookies
+  else if (req.url === "/check-cookies") {
+    res.writeHead(200, { "Content-Type": "text/plain" });
+    let cookiepresent = false;
+    let cookiesobj = req.headers.cookie;
+
+    if (cookiesobj == true) {
```

Comment:
Great work, I think this was the part I struggled with, and your implementation is easy to understand.